### DISCUSSION TITLE
Dependency Parser Understanding

### DISCUSSION THREAD

**Discussion by ElisonSherton** [explosion/spaCy#8645] [Help: Other Questions] [answered]
Hey everyone,

Good Day! 

I am trying to understand dependency parser from spacy and how it works. Consider the following example

![image](https://user-images.githubusercontent.com/45599762/124921625-36e7df00-e016-11eb-8f3a-62d48dada3a9.png)

In this, isn't `free flow` something which modifies the noun `sleeves` ; so shouldn't there be connections from sleeve to free and sleeves to flow in the graph? 

Also, had a similar question which I had posted on [prodigy forums](https://support.prodi.gy/t/understanding-spacys-dependency-parser-better/4265) where it was suggested this was a better place to host such a discussion.

I am a novice in NLP and would appreciate it if someone could help me or point me to the right resources wrt how dependency parsing works.

Thanks & Regards,
Vinayak.

**Comment by DavidGerva** [comment, marked as answer]
Hi Elison,

Let's see if you obtain some more useful answer but I think the answer is quite simple: models can be wrong, it depends on "what they are trained on".
In fact, the same sentences, with the large model, gives these results (probably the one you expect, at least in the first case!):
![image](https://user-images.githubusercontent.com/1856249/124933007-d18af280-e003-11eb-8ed4-cbbb15bff489.png)
![image](https://user-images.githubusercontent.com/1856249/124933682-5d048380-e004-11eb-8a72-873d914911fc.png)

You cannot try the large model online, but You can on your pc with this code:

    import spacy

    if __name__ == "__main__":
        nlp = spacy.load('en_core_web_lg')
        doc = nlp(u"Perfect for balmy, vibrant summer days")
        spacy.displacy.serve(doc, style='dep')

**Reply by polm** [reply]
This is a great explanation!

I will just add that the spaCy models are mainly trained on newspaper articles, while it looks like this sentence (and the one on the Prodigy forums) are ecommerce product descriptions with somewhat flowery language. So while I would expect the models to mostly work, especially the larger ones, this language is different than what they're trained on.
